### PR TITLE
Adding documentation for schedules APIs

### PR DESCRIPTION
**Pull request by IaroslavTitov**
## Description
- Adding documentation for schedules APIs

## Checklist:

- [ ] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 1]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
     "AWS_SESSION_TOKEN": "<redacted>""
 ```
 
+## Schedules
+
+{{< notes >}}
+Pulumi Schedules REST API endpoints are currently in private preview.
+{{< /notes >}}
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2536]: When we publish these docs that won't be true anymore.

```suggestion
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 2]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+```
+
+#### Parameters
+
+| Parameter           | Type   | In    | Description                                                                                                  |
+|---------------------|--------|-------|--------------------------------------------------------------------------------------------------------------|
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `timestamp`         | string | body  | ISO timestamp of when to destroy the stack                                                                   |
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2668]: Can we put an example timestamp here

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 3]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+
+#### Example
+
+```bash
+curl \
+  -H "Accept: application/vnd.pulumi+8" \
+  -H "Content-Type: application/json" \
+  -H "Authorization: token $PULUMI_ACCESS_TOKEN" \
+  --request POST \
+  --data '{"timestamp":"{timestampValue}","deleteAfterDestroy":"{deleteAfterDestroyValue}"}' \
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2679]: Make it a real example, so fill in the values instead of `{timestampValue}` and `{deleteAfterDestroyValue}`. Applies throughout.

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md (hunk 2 above)
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2668]:
```suggestion
| `timestamp`         | string | body  | ISO 8601 timestamp specifying when to destroy the stack                                                                   |
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 4]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+        }
+    }
+  },
+  "created": "2024-04-19 01:00:00.000",
+  "modified": "2024-04-19 01:00:00.000",
+  "lastExecuted": null
+}
+```
+
+### Create Raw Operation Schedule
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2715]:
```suggestion
### Create Raw Deployment Schedule
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 5]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+
+#### Parameters
+
+| Parameter           | Type   | In    | Description                                                                                                  |
+|---------------------|--------|-------|--------------------------------------------------------------------------------------------------------------|
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `timestamp`         | string | body  | ISO timestamp of when to destroy the stack                                                                   |
+| `deleteAfterDestroy`| bool   | body  | whether to delete the stack after resources are destroyed                                                    |
```
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2669]:
```suggestion
| `deleteAfterDestroy`| bool   | body  | true if the stack should be deleted after resources are destroyed                                                    |
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 6]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+
+#### Parameters
+
+| Parameter           | Type   | In    | Description                                                                                                  |
+|---------------------|--------|-------|--------------------------------------------------------------------------------------------------------------|
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `scheduleCron`      | string | body  | cron expression for when to run drift detection                                                              |
+| `autoRemediate`     | bool   | body  | whether to remediate any detected drift                                                                      |
```
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2609]:
```suggestion
| `autoRemediate`     | bool   | body  | true if detected drift should be remediated automatically                                                                     |
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 7]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+|---------------------|--------|-------|--------------------------------------------------------------------------------------------------------------|
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `scheduleCron`      | string | body  | cron expression for when to run the pulumi operation                                                         |
+| `scheduleOnce`      | string | body  | ISO timestamp of when to destroy the stack                                                                   |
+| `operation`         | string | body  | pulumi operation to perform. Can be "update", "destroy", "refresh" or "preview"                              |
+
+Note: only pass in either `scheduleCron` or `scheduleOnce`, and not both. One of them is required though.
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2732]:
```suggestion
Note: Only one of `scheduleCron` and `scheduleOnce` must be set.
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 8]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+
+```
+Status: 200 OK
+```
+
+```
+{
+  "id": "12345678-8102-447f-b246-e9ec85786e23",
+  "orgID": "87654321-8b3b-418a-8c01-5ddd0e00bace",
+  "scheduleCron": "0 0 * * *",
```
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2693]: This looks wrong to me. I think `scheduleOnce` should be set instead.

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 9]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+
+#### Parameters
+
+| Parameter           | Type   | In    | Description                                                                                                  |
+|---------------------|--------|-------|--------------------------------------------------------------------------------------------------------------|
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `scheduleCron`      | string | body  | cron expression for when to run the pulumi operation                                                         |
+| `scheduleOnce`      | string | body  | ISO timestamp of when to destroy the stack                                                                   |
```
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2729]:
```suggestion
| `scheduleOnce`      | string | body  | ISO 8601 timestamp specifying when to destroy the stack                                                                   |
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 10]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+curl \
+  -H "Accept: application/vnd.pulumi+8" \
+  -H "Content-Type: application/json" \
+  -H "Authorization: token $PULUMI_ACCESS_TOKEN" \
+  --request POST \
+  --data '{ "scheduleCron":"{scheduleCronValue}", "request": { "operation": "{operationValue}" } }' \
+  https://api.pulumi.com/api/stacks/{organization}/{project}/{stack}/deployments/schedules
+```
+
+The `request` field is really a [CreateDeploymentRequest](/docs/pulumi-cloud/deployments/api/#create-deployment) under the hood. You can pass in the same parameters to customize the scheduled deployment, but they are not required.
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2746]: Remove this section since we've explained the usage in the parameters section.
```suggestion
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md (hunk 7 above)
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2732]:
```suggestion
Note: Exactly one of `scheduleCron` and `scheduleOnce` must be set.
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 11]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+|---------------------|--------|-------|--------------------------------------------------------------------------------------------------------------|
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `scheduleCron`      | string | body  | cron expression for when to run the pulumi operation                                                         |
+| `scheduleOnce`      | string | body  | ISO timestamp of when to destroy the stack                                                                   |
+| `operation`         | string | body  | pulumi operation to perform. Can be "update", "destroy", "refresh" or "preview"                              |
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2730]: This should be `request` instead of `operation` I think?
```suggestion
| `request`         | [CreateDeploymentRequest](/docs/pulumi-cloud/deployments/api/#create-deployment) | body  | The create deployment request object that will be executed on every invocation |
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md (hunk 10 above)
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2746]:
```suggestion
The `request` field is a [CreateDeploymentRequest](/docs/pulumi-cloud/deployments/api/#create-deployment), which optionally allows full customization of the deployment to be scheduled.
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 12]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+|---------------------|--------|-------|--------------------------------------------------------------------------------------------------------------|
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `scheduleID`        | string | path  | schedule ID that you want to update                                                                          |
+| `scheduleCron`      | string | body  | cron expression for when to run drift detection                                                              |
+| `autoRemediate`     | bool   | body  | whether to remediate any detected drift                                                                      |
```
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2837]:
```suggestion
| `autoRemediate`     | bool   | body  | true if detected drift should be remediated automatically                                                                      |
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 13]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+|---------------------|--------|-------|--------------------------------------------------------------------------------------------------------------|
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `scheduleID`        | string | path  | schedule ID that you want to update                                                                          |
+| `timestamp`         | string | body  | ISO timestamp of when to destroy the stack                                                                   |
+| `deleteAfterDestroy`| bool   | body  | whether to delete the stack after resources are destroyed                                                    |
```
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2896]:
```suggestion
| `deleteAfterDestroy`| bool   | body  | true if the stack should be deleted after resources are destroyed                                                    |
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 14]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+        }
+    }
+  },
+  "created": "2024-04-19 01:00:00.000",
+  "modified": "2024-04-19 01:00:00.000",
+  "lastExecuted": null
+}
+```
+
+### Update Raw Operation Schedule
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2942]:
```suggestion
### Update Raw Deployment Schedule
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 15]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+|---------------------|--------|-------|--------------------------------------------------------------------------------------------------------------|
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `scheduleID`        | string | path  | schedule ID that you want to update                                                                          |
+| `scheduleCron`      | string | body  | cron expression for when to run the pulumi operation                                                         |
+| `scheduleOnce`      | string | body  | ISO timestamp of when to destroy the stack                                                                   |
```
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2957]:
```suggestion
| `scheduleOnce`      | string | body  | ISO 8601 timestamp specifying when to destroy the stack                                                                   |
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 16]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+curl \
+  -H "Accept: application/vnd.pulumi+8" \
+  -H "Content-Type: application/json" \
+  -H "Authorization: token $PULUMI_ACCESS_TOKEN" \
+  --request POST \
+  --data '{ "scheduleCron":"{scheduleCronValue}", "request": { "operation": "{operationValue}" } }' \
+  https://api.pulumi.com/api/stacks/{organization}/{project}/{stack}/deployments/schedules/{scheduleID}
+```
+
+The `request` field is really a [CreateDeploymentRequest](/docs/pulumi-cloud/deployments/api/#create-deployment) under the hood. You can pass in the same parameters to customize the scheduled deployment, but they are not required.
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2974]: Same feedback as in Create Raw Deployment Schedule

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 17]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `scheduleID`        | string | path  | schedule ID that you want to update                                                                          |
+| `scheduleCron`      | string | body  | cron expression for when to run the pulumi operation                                                         |
+| `scheduleOnce`      | string | body  | ISO timestamp of when to destroy the stack                                                                   |
+| `operation`         | string | body  | pulumi operation to perform. Can be "update", "destroy", "refresh" or "preview"                              |
+
+Note: only pass in either `scheduleCron` or `scheduleOnce`, and not both. One of them is required though.
```
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2960]:
```suggestion
Note: Exactly one of `scheduleCron` and `scheduleOnce` must be set.
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md (hunk 16 above)
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2974]:
```suggestion
The `request` field is a [CreateDeploymentRequest](/docs/pulumi-cloud/deployments/api/#create-deployment), which optionally allows full customization of the deployment to be scheduled.
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 18]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+```
+DELETE /api/stacks/{organization}/{project}/{stack}/deployments/schedules/{scheduleID}
+```
+
+| Parameter           | Type   | In    | Description                                                                                                  |
+|---------------------|--------|-------|--------------------------------------------------------------------------------------------------------------|
+| `organization`      | string | path  | organization name                                                                                            |
+| `project`           | string | path  | project name                                                                                                 |
+| `stack`             | string | path  | stack name                                                                                                   |
+| `scheduleID`        | string | path  | schedule ID that you want to get                                                                             |
```
Comment (lblackstone) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:3013]:
```suggestion
| `scheduleID`        | string | path  | schedule ID that you want to delete                                                                             |
```

**Review by lblackstone — CHANGES_REQUESTED** [review]
A few minor suggestions, but looks great overall!

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 19]
```diff
@@ -2529,6 +2529,610 @@ Status: 200 OK
...
+      "modified": "2024-04-19 01:00:00.000",
+      "lastExecuted": null
+    }
+  ]
+}
+
+```
+
+### Create Drift Schedule
+
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2596]: Would be great to add a note here and Update Drift Schedule and Create/Update TTL Schedule that says something like:

```
This is a convenience API for ease of use to create/update [this] schedule, but the same configuration is possible via the raw deployments schedule APIs. 
```

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md (hunk 8 above)
Comment (IaroslavTitov) [reply themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2693]: That is correct, great catch! Fixed here and in Update TTL

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md (hunk 3 above)
Comment (IaroslavTitov) [reply themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2679]: Throughout the doc, I saw examples of both and wasn't sure which one was best. Updating to real values.

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md (hunk 11 above)
Comment (IaroslavTitov) [reply themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2730]: I was aiming to make it simpler for users, and to match UI, but yeah, ideally we should point them to full range of things for the raw one, updating to this.

File: themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md [hunk 20]
```diff
@@ -2529,6 +2529,614 @@ Status: 200 OK
     "AWS_SESSION_TOKEN": "<redacted>""
 ```
 
+## Schedules
+
+{{< notes >}}
+Pulumi Schedules REST API endpoints are currently in public preview.
+{{< /notes >}}
```
Comment (komalali) [inline themes/default/content/docs/pulumi-cloud/cloud-rest-api/_index.md:2536]: Remove this please!
```suggestion
```

**Review by komalali — COMMENTED** [review]
Screenshot 2024-04-23 at 1 44 43 PM

The current order is a little bit hard to follow as the endpoints I expect to be together are not quite.
Can we please reorder the endpoints as follows:
- Create Drift Schedule
- Update Drift Schedule
- Create TTL Schedule
- Update TTL Schedule
- Create Raw Deployment Schedule
- Update Raw Deployment Schedule
- Get Schedule
- Delete Schedule
- Pause Schedule
- Resume Schedule
- List Schedules
- List Scheduled Deployment History

**Comment by komalali** [comment]
Just a couple ordering/deletion changes and we should be good to go! Thanks for all your work on this @IaroslavTitov!

**Comment by IaroslavTitov** [comment]
Closing this in favor of https://github.com/pulumi/pulumi-hugo-private/pull/203 because we need to release this via hugo-private
